### PR TITLE
Remove CacheRequestDetails

### LINKED ISSUE #10904: Remove CacheRequestDetails
It looks like all the use cases of it can be replaced by `Request` instead, making `CacheRequestDetails` unnecessary.

Code: `components/net/fetch/cors_cache.rs`

### PR DESCRIPTION
**Pull request by shssoichiro**
Fixes #10904 

Note that the related web-platform-tests failed both before and after this commit

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10971)

### CODE REVIEW

**Comment by jdm** [comment]
Looks like the unit tests need updating too: `./mach test-unit -p net`.

**Comment by shssoichiro** [comment]
Pushed new commit with updated unit tests

File: components/net/fetch/cors_cache.rs [hunk 1]
```diff
@@ -122,21 +116,21 @@ impl CORSCache {
 
     /// Returns true if an entry with a
     /// [matching header](https://fetch.spec.whatwg.org/#concept-cache-match-header) is found
-    pub fn match_header(&mut self, request: CacheRequestDetails, header_name: &str) -> bool {
+    pub fn match_header(&mut self, request: Request, header_name: &str) -> bool {
```
Comment (KiChjang) [inline components/net/fetch/cors_cache.rs]: Why is this changing the ownership of request? Can it not use a reference instead?

File: components/net/fetch/cors_cache.rs [hunk 2]
```diff
@@ -122,21 +116,21 @@ impl CORSCache {
...
         self.find_entry_by_header(&request, header_name).is_some()
     }
 
     /// Updates max age if an entry for a
     /// [matching header](https://fetch.spec.whatwg.org/#concept-cache-match-header) is found.
     ///
     /// If not, it will insert an equivalent entry
-    pub fn match_header_and_update(&mut self, request: CacheRequestDetails,
+    pub fn match_header_and_update(&mut self, request: Request,
```
Comment (KiChjang) [inline components/net/fetch/cors_cache.rs]: Same here, why is this a move on request?

File: components/net/fetch/cors_cache.rs [hunk 3]
```diff
@@ -145,20 +139,21 @@ impl CORSCache {
 
     /// Returns true if an entry with a
     /// [matching method](https://fetch.spec.whatwg.org/#concept-cache-match-method) is found
-    pub fn match_method(&mut self, request: CacheRequestDetails, method: Method) -> bool {
+    pub fn match_method(&mut self, request: Request, method: Method) -> bool {
```
Comment (KiChjang) [inline components/net/fetch/cors_cache.rs]: Ditto about moving.

File: components/net/fetch/cors_cache.rs [hunk 4]
```diff
@@ -145,20 +139,21 @@ impl CORSCache {
...
         self.find_entry_by_method(&request, method).is_some()
     }
 
     /// Updates max age if an entry for
     /// [a matching method](https://fetch.spec.whatwg.org/#concept-cache-match-method) is found.
     ///
     /// If not, it will insert an equivalent entry
-    pub fn match_method_and_update(&mut self, request: CacheRequestDetails, method: Method, new_max_age: u32) -> bool {
+    pub fn match_method_and_update(&mut self, request: Request, method: Method, new_max_age: u32) -> bool {
```
Comment (KiChjang) [inline components/net/fetch/cors_cache.rs]: Ditto about moving.

File: components/net/fetch/cors_cache.rs (hunk 1 above)
Comment (shssoichiro) [inline components/net/fetch/cors_cache.rs]: This is simply mimicking the previous behavior. I thought it was better to use a reference as well but didn't want to overstep my bounds. I can push a fix later on for these.

**Comment by KiChjang** [comment]
Speaking of overstepping your bounds, I'd appreciate it if you left a comment claiming the issue next time you work on it.

**Comment by shssoichiro** [comment]
I apologize. I'll do so next time.

I've pushed a new commit switching the moves to references. Let me know if all looks good and I can squash.

File: components/net/fetch/methods.rs [hunk 5]
```diff
@@ -402,13 +402,13 @@ fn http_fetch(request: Rc<Request>,
 
         // Substep 1
         if cors_preflight_flag {
-            let method_cache_match = cache.match_method(request.as_ref().clone(),
+            let method_cache_match = cache.match_method(request.as_ref(),
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: This could just be &request.

File: components/net/fetch/methods.rs [hunk 6]
```diff
@@ -402,13 +402,13 @@ fn http_fetch(request: Rc<Request>,
...
                                                         request.method.borrow().clone());
 
             let method_mismatch = !method_cache_match && (!is_simple_method(&request.method.borrow()) ||
                                                           request.use_cors_preflight);
             let header_mismatch = request.headers.borrow().iter().any(|view|
-                !cache.match_header(request.as_ref().clone(), view.name()) && !is_simple_header(&view)
+                !cache.match_header(request.as_ref(), view.name()) && !is_simple_header(&view)
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: Same here.

File: components/net/fetch/methods.rs [hunk 7]
```diff
@@ -402,23 +402,13 @@ fn http_fetch(request: Rc<Request>,
...
         if cors_preflight_flag {
-            let origin = request.origin.borrow().clone();
-            let url = request.current_url();
-            let credentials = request.credentials_mode == CredentialsMode::Include;
-            let method_cache_match = cache.match_method(CacheRequestDetails {
-                origin: origin.clone(),
-                destination: url.clone(),
-                credentials: credentials
-            }, request.method.borrow().clone());
+            let method_cache_match = cache.match_method(request.as_ref(),
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: This can just be &*request.

File: components/net/fetch/methods.rs [hunk 8]
```diff
@@ -402,23 +402,13 @@ fn http_fetch(request: Rc<Request>,
...
 
             let method_mismatch = !method_cache_match && (!is_simple_method(&request.method.borrow()) ||
                                                           request.use_cors_preflight);
             let header_mismatch = request.headers.borrow().iter().any(|view|
-                !cache.match_header(CacheRequestDetails {
-                    origin: origin.clone(),
-                    destination: url.clone(),
-                    credentials: credentials
-                }, view.name()) && !is_simple_header(&view)
+                !cache.match_header(request.as_ref(), view.name()) && !is_simple_header(&view)
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: Same here.

File: components/net/fetch/methods.rs [hunk 9]
```diff
@@ -1027,20 +1017,12 @@ fn cors_preflight_fetch(request: Rc<Request>, cache: &mut CORSCache) -> Response
 
         // Substep 11, 12
         for method in &methods {
-            cache.match_method_and_update(CacheRequestDetails {
-                origin: request.origin.borrow().clone(),
-                destination: request.current_url(),
-                credentials: request.credentials_mode == CredentialsMode::Include
-            }, method.clone(), max_age);
+            cache.match_method_and_update(request.as_ref(), method.clone(), max_age);
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: Same here.

File: components/net/fetch/methods.rs [hunk 10]
```diff
@@ -1027,20 +1017,12 @@ fn cors_preflight_fetch(request: Rc<Request>, cache: &mut CORSCache) -> Response
...
         }
 
         // Substep 13, 14
         for header_name in &header_names {
-            cache.match_header_and_update(CacheRequestDetails {
-                origin: request.origin.borrow().clone(),
-                destination: request.current_url(),
-                credentials: request.credentials_mode == CredentialsMode::Include
-            }, &*header_name, max_age);
+            cache.match_header_and_update(request.as_ref(), &*header_name, max_age);
```
Comment (KiChjang) [inline components/net/fetch/methods.rs]: Same here.

File: tests/unit/net/fetch.rs [hunk 11]
```diff
@@ -248,11 +248,8 @@ fn test_cors_preflight_cache_fetch() {
     assert_eq!(1, counter.load(Ordering::SeqCst));
 
     // The entry exists in the CORS-preflight cache
-    assert_eq!(true, cache.match_method(CacheRequestDetails {
-        origin: origin,
-        destination: url,
-        credentials: false
-    }, Method::Get));
+    assert_eq!(true, cache.match_method(wrapped_request0.as_ref(), Method::Get));
+    assert_eq!(true, cache.match_method(wrapped_request1.as_ref(), Method::Get));
```
Comment (KiChjang) [inline tests/unit/net/fetch.rs]: `&*wrapped_request`

**Comment by KiChjang** [comment]
Squash!

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Pull request merged**